### DISCUSSION TITLE
How to unversion a versioned formula

### DISCUSSION THREAD

**Discussion by sidekick-eimantas** [Homebrew/discussions#3888] [Tap maintenance and Homebrew development] [answered]
Hi

We're trying to set up an unversioned formula clone of the versioned [python@3.10](https://github.com/Homebrew/homebrew-core/blob/master/Formula/python@3.10.rb) formula, in our in-house homebrew tap.
Am I missing anything, or is unversioning these formulas as simple as removing the suffix from the .rb file and from the ruby class name?

Thank you

**Comment by p-linnane** [comment, marked as answer]
There are a number of issues at play. `python3` is currently aliased to `python@3.10` but that will eventually change. You'll need to remove the `bottle` stanza and build from source. You will need to manually update the `resource` & `patch` blocks as appropriate. Give it a shot and make sure you test it plenty.

**Reply by sidekick-eimantas** [reply]
Hi @p-linnane, thanks for looking at this

I may have not explained what I'm trying to do correctly, as I don't fully understand your answer, specifically around why `python3` currently being aliased to `python@3.10` is a problem.
I want `brew install sidekick-eimantas/custom/python` to install what is currently in the `python@3.10` formula, such that `python --version`, after install, would output v3.10.x (v3.10.8 as of the time writing this)

I've attempted the naive approach of just removing the `@3.10` suffix from the .rb file and the class name in the `.rb` file, which resulted in brew hitting 404's when fetching metadata for the bottle. So I'm guessing it's not as simple as that.

**Reply by SMillerDev** [reply]
Well, obviously you can't rename something and still expect bottle support with the new name. You'll have to make your own bottles in your tap. Additionally you could misuse the tap_migrations file and say that `python` migrated to `Homebrew/core/python@3.10`.

**Reply by sidekick-eimantas** [reply]
Hi @SMillerDev, thanks for the help

Unfortunately, it's not obvious to me why renaming the formula would break the bottles. Could you elaborate? Or maybe there's documentation you could point me to that explains this or how bottles work?

Thank you

**Reply by SMillerDev** [reply]
Well, bottles are precompiled binaries that are hosted on the taps GitHub packages by default. Changing the name changes the identifying information for the formula so it would point to a `python` bottle now, which the checksum would never match, and on top of that it's no longer in Homebrew/core so you'll have to upload the bottles yourself or let users compile from source.

https://docs.brew.sh/Bottles

**Reply by sidekick-eimantas** [reply]
Ah, I see. I think this gives me enough info to figure this out.
Thanks to both of you !
